contracts/stake-cw20/src/contract.rs

```diff
@@ -278,6 +302,38 @@ pub fn execute_fund(
         .add_attribute("amount", amount))
 }
 
+pub fn execute_add_hook(
+    deps: DepsMut,
+    _env: Env,
+    info: MessageInfo,
+    addr: Addr,
+) -> Result<Response, ContractError> {
+    let config: Config = CONFIG.load(deps.storage)?;
+    if config.owner != Some(info.sender.clone()) && config.manager != Some(info.sender) {
+        return Err(ContractError::Unauthorized {});
+    };
+    HOOKS.add_hook(deps.storage, addr.clone())?;
```

Comment:
Does `HOOKS` validate this address?

Reply:
We should also be careful here because a valid address must be a contract address.

Reply:
is the addr type validated when it is parsed from Json?

Reply:
Nope :/ https://docs.rs/cosmwasm-std/latest/cosmwasm_std/struct.Addr.html

Reply:
cool added validation

Reply:
ty for looking into it